Argo/Types/StandardTypes.swift

```diff
@@ -69,10 +69,10 @@ public extension Array where Element: Decodable, Element == Element.DecodedType
   }
 }
 
-public extension Dictionary where Value: Decodable, Value == Value.DecodedType {
-  static func decode(j: JSON) -> Decoded<[String: Value]> {
+public extension CollectionType where Self: DictionaryLiteralConvertible, Self.Value: Decodable, Self.Value == Self.Value.DecodedType {
```

Comment:
I could be wrong but I think we can only take advantage of the implicit `Self.` when adding constraints to types that directly belong to the type or protocol that is being extended (in this case `CollectionType`), and not when adding constraints to types that belong to another constraint on the extended type. Does that make sense?

Because we're adding `DictionaryLiteralConvertible` as a separate constraint on `Self`, the `Self.` is necessary to constrain types belonging to `DictionaryLiteralConvertible.